Model/Source/DefaultOptions.php

```diff
@@ -140,33 +157,95 @@ public function getDefaultLargeFormat(string $option): bool
     }
 
     /**
-     * @param string $option
+     * @param  string $carrier
+     * @param  string $option
      *
      * @return bool
      */
-    public function getDefaultOptionsWithoutPrice(string $option): bool
+    public function hasDefaultOptionsWithoutPrice(string $carrier, string $option): bool
     {
-        $settings = self::$helper->getStandardConfig('default_options');
+        $settings = self::$helper->getStandardConfig($carrier, 'default_options');
 
-        return $settings[$option . '_active'] === '1';
+        return '1' === ($settings[$option . '_active'] ?? null);
     }
 
     /**
      * Get default value of insurance based on order grand total
      *
+     * @param  string $carrier
+     *
      * @return int
      */
-    public function getDefaultInsurance()
+    public function getDefaultInsurance(string $carrier): int
     {
-        if ($this->getDefault('insurance_500')) {
+        $shippingAddress = self::$order->getShippingAddress();
+        $shippingCountry = $shippingAddress ? $shippingAddress->getCountryId() : AbstractConsignment::CC_NL;
+
+        if (AbstractConsignment::CC_NL === $shippingCountry) {
+            return $this->getDefaultLocalInsurance($carrier);
+        }
+
+        if (AbstractConsignment::CC_BE === $shippingCountry) {
+            return $this->getDefaultBeInsurance($carrier);
+        }
+
+        return $this->getDefaultEuInsurance($carrier);
+    }
+
+    /**
+     * @param  string $carrier
+     *
+     * @return int
+     */
+    private function getDefaultEuInsurance(string $carrier): int
+    {
+        if ($this->hasDefault(self::INSURANCE_EU_AMOUNT_500, $carrier)) {
             return 500;
         }
 
-        if ($this->getDefault('insurance_250')) {
+        if ($this->hasDefault(self::INSURANCE_EU_AMOUNT_50, $carrier)) {
+            return 50;
+        }
+
+        return 0;
+    }
+
+    /**
+     * @param  string $carrier
+     *
+     * @return int
+     */
+    private function getDefaultBeInsurance(string $carrier): int
```

Comment:
In de BE plugin is BeInsurance de LocalInsurance, ik weet niet of er aparte tarieven zijn naar NL?